### DISCUSSION TITLE
Livewire\Exceptions\MethodNotFoundException: Unable to call component method. Public method [data] not found on component: [usernotnull.toast.http.livewire.toast-component]

### DISCUSSION THREAD

**Discussion by shawndreck** [usernotnull/tall-toasts#237] [Q&A] [answered]
I installed the package locally and it works fine. But I get this error only on production. And I can't figure out why

Local environment. 
Laravel 9.11.0
PHP 8.0

Production 
Laravel 9.11.0
PHP 8.0

I followed through with the installation steps on the package page.

Any clue or hint will help alot.

**Comment by usernotnull** [comment]
Very little is given, so can't reproduce the error to help. 

The lib was tested in many prod envs with no issues. The environment shouldn't be an issue per se, but maybe you can diagnose what changes in your setup between local and prod env.

**Reply by shawndreck** [reply]
You are definitely. I didn't think the lib was the culprit. I figured it might be one of those "gotcha" traps that I may be forgetting.
And it was! It was related to NGINX caching.

I was caching js on the production server, causing this issue. I had a similar problem with Livewire image upload!
Not caching the lib js files with Nginx solved it. 
nginx/conf.d/app.conf (in my case, I named the file  app.conf )
```

server{
    listen 80;
    server_name example.localhost;
    ...

     # Avoid caching livewire.js files
        location ~* livewire.js {
                index index.php index.html index.htm;
                try_files $uri $uri/ /index.php?$query_string;
                gzip_static on;
        }

        # Avoid caching of tall-toast js files

        location ~* tall-toasts.js {
                index index.php index.html index.htm;
                try_files $uri $uri/ /index.php?$query_string;
                gzip_static on;
        }

        location ~* livewire/.* {
                index index.php index.html index.htm;
                try_files $uri $uri/ /index.php?$query_string;
                gzip_static on;
        }

        location ~* toasts.* {
                index index.php index.html index.htm;
                try_files $uri $uri/ /index.php?$query_string;
                gzip_static on;
        }
}

```

**Reply by usernotnull** [reply]
Great 👍 

You could remove the '.js' because you later have '.*'. You could also group to something as below if I'm not mistaken:

```bash
location ~* .(livewire|tall-toast)$  {
```

I remember having a similar discussion with a similar answer related to server config. I will group all questions and lump them in the README in the future.

For now if this resolved your issue kindly mark as resolved. Cheers 👋

**Reply by shawndreck** [reply]
Having these gotcha's in the README will definitely some headaches. 
Thanks for the tip!

**Comment by shawndreck** [comment, marked as answer]
This is a summary of the discussion above as answer

```

server{
    listen 80;
    server_name example.localhost;

    # Other Nginx config here

     # Avoid caching livewire.js and tall toast files
      location ~* .(livewire|tall-toast)$  {
                index index.php index.html index.htm;
                try_files $uri $uri/ /index.php?$query_string;
                gzip_static on;
        }
}
```

**Reply by fdordovlatov** [reply]
that is not good solution because usually a lot of people are using "artisan serve" and "vite" , they can't set this config anywhere

**Reply by usernotnull** [reply]
> that is not good solution because usually a lot of people are using "artisan serve" and "vite" , they can't set this config anywhere

You should not have this issue in these environments, if you do then your problem is something else.
